google/cloud/internal/default_completion_queue_impl.cc

```diff
@@ -217,7 +217,7 @@ DefaultCompletionQueueImpl::MakeDeadlineTimer(
     std::chrono::system_clock::time_point deadline) {
   auto p = AsyncTimerFuture::Create();
   auto op = std::move(p.first);
-  StartOperation(op, [&](void* tag) { op->Set(cq(), deadline, tag); });
+  StartOperation(op, [&](void* tag) { op->Set(*cq(), deadline, tag); });
```

Comment:
Maybe add a comment indicating this indirection is safe, as the value is returned from this class?  Should we mark the class `final` to make sure nobody overrides `cq()` to return `nullptr`?